mobile/src/main/java/org/openhab/habdroid/util/HttpClient.kt

```diff
@@ -40,20 +40,36 @@ import java.util.concurrent.TimeUnit
 import kotlin.coroutines.resume
 import kotlin.coroutines.resumeWithException
 
-class HttpClient constructor(private val client: OkHttpClient, baseUrl: String?, username: String?, password: String?) {
+class HttpClient constructor(client: OkHttpClient, baseUrl: String?, username: String?, password: String?) {
+    private val client: OkHttpClient
     private val baseUrl: HttpUrl? = baseUrl?.toHttpUrlOrNull()
     @VisibleForTesting val authHeader: String? = if (!username.isNullOrEmpty() && !password.isNullOrEmpty())
         Credentials.basic(username, password, StandardCharsets.UTF_8) else null
 
+    init {
+        val clientBuilder = client.newBuilder()
+        if (authHeader != null) {
+            // Forcibly put authorization info into request, as redirect/retry interceptor might have removed it
+            clientBuilder.addNetworkInterceptor { chain ->
+                val request = chain.request()
+                    .newBuilder()
+                    .addHeader("Authorization", authHeader)
+                    .build()
+                chain.proceed(request)
+            }
+        }
+        this.client = clientBuilder.build()
+    }
     enum class CachingMode {
         DEFAULT,
         AVOID_CACHE,
         FORCE_CACHE_IF_POSSIBLE
     }
 
     fun makeSse(url: HttpUrl, listener: EventSourceListener): EventSource {
-        val request = makeAuthenticatedRequestBuilder()
+        val request = Request.Builder()
             .url(url)
+            .addHeader("User-Agent", "openHAB client for Android")
```

Comment:
I'd make constants out of those two strings and "Authorization" above.

Reply:
Is it really that useful? I'm not sure about it. If you think we really should do it, I think I'll change it back to a common method (`makeRequestBuilder`).
For the `Authorization` string, I don't think making it a constant makes sense, as it's only used once.

Reply:
"Authorization" is also used in the MainActivity. I'd put it in constants as we need to use these headers for the video widgets as well (#1829) and there we cannot use a common function.

Reply:
I'd keep it as is, following the example of the okhttp internal code ¯\_(ツ)_/¯ Since those header names are well-known names in the HTTP spec, IMHO putting them behind constants makes the code harder to (immediately) understand.